src/trcb.erl

```diff
@@ -23,11 +23,40 @@
 
 -include("ishikawa.hrl").
 
+-export([causal_delivery/3, try_to_deliever/2]).
+
 %% Broadcast message.
 -callback tcbcast(message()) -> ok.
 
 %% Deliver a message.
--callback tcbdeliver(message(), timestamp()) -> ok.
+-callback tcbdeliver(actor(), message(), timestamp()) -> ok.
 
 %% Determine if a timestamp is stable.
 -callback tcbstable(timestamp()) -> {ok, boolean()}.
+
+%% check if a message should be deliver and deliver it, if not add it to the queue
+-spec causal_delivery({actor(), message(), timestamp()}, timestamp(), [{actor(), message(), timestamp()}]) -> {timestamp(), [{actor(), message(), timestamp()}]}.
+causal_delivery({Origin, Msg, MsgVV}, VV, Queue) ->
+    case vclock:dominates(MsgVV, VV) of
+        true ->
+            NewVV = vclock:increment(Origin, VV),
+            %% TODO: Handle message,
+            try_to_deliever(Queue, {NewVV, Queue});
+        false ->
+            {VV, [Queue | [{Origin, Msg, MsgVV}]]}
```

Comment:
I think this is another case where the list concatenation is not working correctly.  

Did you mean this instead?

```Queue ++ [{Origin, Msg, MsgVV}]```